Source/CarthageKit/Project.swift

```diff
@@ -559,6 +598,25 @@ private func frameworksInDirectory(directoryURL: NSURL) -> SignalProducer<NSURL,
 		}
 }
 
+/// Sends the URL to each dSYM found in the given directory
+private func dSYMsInDirectory(directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+	return filesInDirectory(directoryURL, "com.apple.xcode.dsym")
+}
+
+/// Sends the URL of the dSYM whose UUIDs match those of the given framework, or
+/// errors if there was an error parsing a dSYM contained within the directory.
+private func dSYMForFramework(frameworkURL: NSURL, inDirectoryURL directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+	return combineLatest(UUIDsForFramework(frameworkURL), dSYMsInDirectory(directoryURL))
```

Comment:
Always the last value of `dSYMsInDirectory` is used as `dSYMURL` (in my case it is a dSYM for Mac) because of the following reasons:

- `combineLatest` is used
- `UUIDsForFramework(frameworkURL)` takes some time
- `dSYMsInDirectory` sends all values in the meantime

As a result, coping dSYM for iOS always fails. :cry: